covid_analyzer/utils.py

```diff
@@ -0,0 +1,157 @@
+    total_deaths_around_the_globe = 0
+    for covid_cases_details_per_country in covid_data[CASES_STATS]:
+        if all([covid_cases_details_per_country[COUNTRY_NAME_IDX_IN_COVID_STATS] in countries_following_the_safety_measure,
+                (deaths:=covid_cases_details_per_country[TOTAL_DEATHS]) != EMPTY_STRING]
+            ):
+                total_deaths_in_country_following_the_measures += int(deaths)
+                
+        if (deaths:=covid_cases_details_per_country[TOTAL_DEATHS]) != EMPTY_STRING:
+            total_deaths_around_the_globe += int(deaths)
+
+    countries, death_percentage = (len(countries_following_the_safety_measure),
+           round((total_deaths_in_country_following_the_measures / total_deaths_around_the_globe) * 100, 2))
+    print(f'{death_percentage}% death average found in {countries} countries. ')
+    return countries, death_percentage
+           
+           
+def get_efficiency(covid_data, countries_follwoing_the_safety_measures: list) -> int:
+    """Returns the efficiency against safety measure calculated as 
+        total recovered cases of all countries taken the measure 
+        divided by total cases of all countries taken the measure
+
+    Args:
+        countries_follwoing_the_safety_measures (list): list of countries that follow safety measure
+
+    Returns:
+        int: efficiency value
+    """
+
+    total_recovered_cases_of_all_countries = 0
+    total_cases_of_all_countries = 0
+    for covid_stats_detail_per_country in covid_data[CASES_STATS]:
+        country_name = covid_stats_detail_per_country[COUNTRY_NAME_IDX_IN_COVID_STATS]
+        if all([
+                (country_name != EMPTY_STRING),
+                (country_name in countries_follwoing_the_safety_measures)
+            ]):
+            if all([
+                covid_stats_detail_per_country[TOTAL_CASES] != EMPTY_STRING,
+                covid_stats_detail_per_country[TOTAL_RECOVERED_PATIENTS] != EMPTY_STRING,
+                covid_stats_detail_per_country[TOTAL_RECOVERED_PATIENTS] != ZERO
+                ]):
+                total_cases_of_all_countries += int(covid_stats_detail_per_country[TOTAL_CASES])
+                total_recovered_cases_of_all_countries += int(covid_stats_detail_per_country[TOTAL_RECOVERED_PATIENTS])
+                         
+    if total_cases_of_all_countries == ZERO:
+        return ZERO
+    
+    return round(total_recovered_cases_of_all_countries/total_cases_of_all_countries, 2)
+
+
+def get_most_efficient_safety_measures(covid_data) -> list:
+    """find most efficient safey measures with efficiency value
+
+    Returns:
+        list: top 5 most efficient measures with efficiency value
+    """
+    efficiencies_against_safety_measure = []
+    unique_safety_measures = list(
+            set([
+            safety_measure_details[IMPLEMENTED_SAFETY_MEASURES] for safety_measure_details in covid_data[SAFETY_MEASURES]
+            ])
+    )
```

Comment:
Is there any need to convert a list into a list again?
```suggestion
    unique_safety_measures =set([
        safety_measure_details[IMPLEMENTED_SAFETY_MEASURES] for safety_measure_details in 
        covid_data[SAFETY_MEASURES]
     ])
```

Reply:
set() on some iterable will return set. Therefore, I converted the data type to list!